pkg/storkctl/actionstatus.go

```diff
@@ -0,0 +1,156 @@
+package storkctl
+
+import (
+	"fmt"
+
+	storkv1 "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	storkops "github.com/portworx/sched-ops/k8s/stork"
+	"github.com/spf13/cobra"
+	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubernetes/pkg/printers"
+)
+
+var drActionColumns = []string{"NAME", "CREATED", "STAGE", "STATUS", "ADDITIONAL INFO"}
+
+func newGetFailoverStatusCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	getFailoverCommand := &cobra.Command{
+		Use:   failoverCommand,
+		Short: "Get the status of failover actions",
+		Run: func(c *cobra.Command, args []string) {
+			var actions *storkv1.ActionList
+			var failoverActionList *storkv1.ActionList
+			var err error
+			// user has to provide the namespace from which they want to get list of actions using -n flag
+			namespace := cmdFactory.GetNamespace()
+			if len(args) > 0 {
+				// name of action has been specified
+				actions = new(storkv1.ActionList)
+				for _, actionName := range args {
+					action, err := storkops.Instance().GetAction(actionName, namespace)
+					if err != nil {
+						util.CheckErr(err)
+						return
+					}
+					actions.Items = append(actions.Items, *action)
+				}
+			} else {
+				// fetch all the actions in the given namespace
+				actions, err = storkops.Instance().ListActions(namespace)
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+			}
+
+			// filter it down to actions with actionType failover
+			failoverActionList = new(storkv1.ActionList)
+			for _, action := range actions.Items {
+				if action.Spec.ActionType == storkv1.ActionTypeFailover {
+					failoverActionList.Items = append(failoverActionList.Items, action)
+				}
+			}
+
+			if len(failoverActionList.Items) == 0 {
+				handleEmptyList(ioStreams.Out)
+				return
+			}
+
+			if err := printObjects(c, failoverActionList, cmdFactory, drActionColumns, drActionPrinter, ioStreams.Out); err != nil {
+				util.CheckErr(err)
+				return
+			}
+		},
+	}
+	cmdFactory.BindGetFlags(getFailoverCommand.Flags())
+	return getFailoverCommand
+}
+
+func drActionPrinter(actionList *storkv1.ActionList, options printers.GenerateOptions) ([]metav1beta1.TableRow, error) {
+	if actionList == nil {
+		return nil, nil
+	}
+
+	rows := make([]metav1beta1.TableRow, 0)
+	for _, action := range actionList.Items {
+		var row metav1beta1.TableRow
+		creationTime := toTimeString(action.CreationTimestamp.Time)
+		additionalInfo := action.Status.Reason
+		if action.Status.Summary != nil {
+			if len(action.Status.Summary.FailoverSummaryItem) > 0 {
+				// find out number of successfully scaled up namespaces
+				totalNamespaces := len(action.Status.Summary.FailoverSummaryItem)
+				successfulNamespaces := 0
+				rollbackSuccessfulNamespaces := 0
+				rollbackFailedNamespaces := 0
```

Comment:
Why this count is required?

Reply:
count is required because we need it to print the ratio of successfully scaled up / rolled back namespaces in the additional info message.
I have the three counts to judge whether it's a case of a rollback or scale up. So if either of rollbackSuccessful or rollbackFailed is greater than 0 , that means it's the case of rollback, else it's a normal scale up operation.